src/views/dataView/BikeCounterView.vue

```diff
@@ -18,24 +26,76 @@
 <script>
 import MyTable from 'src/component/shared/MyTable.vue';
 import csvFile from 'src/assets/csv/compteurs.csv';
+import Sort from "../../component/shared/Sort.vue";
+import Sort2 from 'src/component/shared/Sort2.vue';
 
 export default {
   components: {
+    Sort2,
     MyTable,
   },
 
+
   data() {
     return {
       loading: false,
-      csvFile
+      bikeCounterData: csvFile,
+
+      sort: {
+        key: 'ID',
+        direction: 'asc',
+      }
     };
   },
 
   computed: {
     headerRow() {
       return Object.keys(csvFile[0]);
-    }
+    },
+
+    filteredData() {
+      // Clone Array, we dont want to mutate directly the data, we want to mutated a computed value that way the orginal data is gone
+      const filteredData = [...this.bikeCounterData];
+
+      // Sort 
+      /*
+        Sort filteredData here based on Sort.vue > sortArray algo you made
+        using this.sort.key and this.sort.direction as how to sort
+      */
+
+      // Filter/Paginate
+
+      return filteredData;
+    },
+
+    filteredHeader() {
+      return {
+        ID: 'Id',
+        Nom: 'Nom du compteur',
+        Statut: 'Statut', 
+        Annee_implante: 'Annee Implante',
+      };
+    },
+  },
+
+  mounted() {
+    this.loadInitalSort();
   },
+
+  methods: {
+    toggleSort(key) {
+      const newSort = {
+        key,
+        direction: 'asc',
+      };
+
+      if (this.sort.key === key && this.sort.direction === 'asc') {
+        newSort.direction = 'desc';
+      }
+
+      this.sort = newSort;
+    }
```

Comment:
le toggleSort est la pour changer le this.sort dans le data par raport a l’option cliquer. `@toggle-sort="toggleSort"` en haut pour le compoenent <Sort2 ... />